zebra-state/src/service/non_finalized_state.rs

```diff
@@ -340,14 +340,13 @@ impl NonFinalizedState {
 
     /// Returns the [`transparent::Utxo`] pointed to by the given
     /// [`transparent::OutPoint`] if it is present in any chain.
+    ///
+    /// UTXOs are returned regardless of whether they have been spent.
     pub fn any_utxo(&self, outpoint: &transparent::OutPoint) -> Option<transparent::Utxo> {
-        for chain in self.chain_set.iter().rev() {
-            if let Some(utxo) = chain.created_utxos.get(outpoint) {
-                return Some(utxo.utxo.clone());
-            }
-        }
-
-        None
+        self.chain_set
+            .iter()
+            .rev()
+            .find_map(|chain| chain.created_utxo(outpoint))
```

Comment:
The `OutPoint` contains the transaction hash, which includes all the content of the `Utxo`. So there is only ever one `Utxo` per `OutPoint`, even across multiple chains.

Reply:
I meant the scenario when there is the same transaction in two parallel chains, for example when two separate miners produce two separate blocks with some transactions being the same. Does your claim hold in this scenario as well?

Reply:
But as I said, I don't think this scenario is important because it doesn't affect Zebra's behavior. I just wanted to understand what actually happens.

Reply:
> I meant the scenario when there is the same transaction in two parallel chains, for example when two separate miners produce two separate blocks with some transactions being the same. Does your claim hold in this scenario as well?

In this case:
- if the transactions are the same, the `OutPoint`s are the same, and they return the same `Utxo`
- if the transactions are different, the transaction hash or index in the `OutPoint`s are different, even if the `Utxo`s have the same value and spend script

https://github.com/ZcashFoundation/zebra/blob/97fb85dca9d00a717661ea0cbab7ce66b8d23977/zebra-chain/src/transparent.rs#L99-L113

Reply:
Got it, thank you for taking the time to explain it.